src/math/bvh.js

```diff
@@ -0,0 +1,456 @@
+import { Vec3 } from '../math/vec3.js';
+import { BoundingBox } from '../shape/bounding-box.js';
+
+/** @typedef {import('../shape/ray.js').Ray} Ray */
+/** @typedef {import('../math/tri.js').Tri} Tri */
+
+class BVHNode {
+    /**
+     * Create a new BVHNode
+     *
+     * @param {Vec3} [aabbMin] - The bounds of the BVH node
+     * @param {Vec3} [aabbMax] - The bounds of the BVH node
+     * @param {number} [leftFirst] - The index of the left child of the BVH node
+     * @param {number} [triCount] - The number of primitives
+     * @ignore
+     */
+    constructor(aabbMin, aabbMax, leftFirst, triCount) {
```

Comment:
change it to handle default params:
`constructor(aabbMin = new Vec3(), aabbMax = new Vec3(), leftFirst = null, triCount = 0) {`

Reply:
Added